tests/cli/main.cpp

```diff
@@ -1593,3 +1589,215 @@ BOOST_FIXTURE_TEST_CASE(cli_use_authorized_transfer, cli_fixture) {
+         signed_transaction create_acct_tx = con.wallet_api_ptr->create_account_with_brain_key(bki.brain_priv_key, "bob", 
+               "nathan", "nathan", true);
+         // this should cause resync which will import the keys of alice and bob
+         generate_block(app1);
+         // attempt to give bob some bitsahres
+         BOOST_TEST_MESSAGE("Transferring bitshares from Nathan to Bob");
+         signed_transaction transfer_tx = con.wallet_api_ptr->transfer("nathan", "bob", "10000", "1.3.0", 
+               "Here are some CORE token for your new account", true);
+         con.wallet_api_ptr->issue_asset("bob", "5", "BOBCOIN", "Here are your BOBCOINs", true);
+      }
+
+      BOOST_TEST_MESSAGE("Alice has agreed to buy 3 BOBCOIN from Bob for 3 BTS. Alice creates an HTLC");
+      // create an HTLC
+      std::string preimage_string = "My Super Long Secret that is larger than 50 charaters. How do I look?\n";
+      fc::hash160 preimage_md = fc::hash160::hash(preimage_string);
+      std::stringstream ss;
+      for(size_t i = 0; i < preimage_md.data_size(); i++)
+      {
+         char d = preimage_md.data()[i];
+         unsigned char uc = static_cast<unsigned char>(d);
+         ss << std::setfill('0') << std::setw(2) << std::hex << (int)uc;
+      }
+      std::string hash_str = ss.str();
+      BOOST_TEST_MESSAGE("Secret is " + preimage_string + " and hash is " + hash_str);
+      uint32_t timelock = fc::days(1).to_seconds();
+      graphene::chain::signed_transaction result_tx 
+            = con.wallet_api_ptr->htlc_create("alice", "bob", 
+            "3", "1.3.0", "HASH160", hash_str, preimage_string.size(), timelock, "Alice to Bob", true);
+
+      // normally, a wallet would watch block production, and find the transaction. Here, we can cheat:
+      std::string alice_htlc_id_as_string;
+      {
+         BOOST_TEST_MESSAGE("The system is generating a block");
+         graphene::chain::signed_block result_block;
+         BOOST_CHECK(generate_block(app1, result_block));
+
+         // get the ID:
+         htlc_id_type htlc_id = result_block.transactions[result_block.transactions.size()-1].operation_results[0].get<object_id_type>();
+         alice_htlc_id_as_string = (std::string)(object_id_type)htlc_id;
+         BOOST_TEST_MESSAGE("Alice shares the HTLC ID with Bob. The HTLC ID is: " + alice_htlc_id_as_string);
+      }
+
+      // Bob can now look over Alice's HTLC, to see if it is what was agreed to.
+      BOOST_TEST_MESSAGE("Bob retrieves the HTLC Object by ID to examine it.");
+      auto alice_htlc = con.wallet_api_ptr->get_htlc(alice_htlc_id_as_string);
+      BOOST_TEST_MESSAGE("The HTLC Object is: " + fc::json::to_pretty_string(alice_htlc));
+
+      // Bob likes what he sees, so he creates an HTLC, using the info he retrieved from Alice's HTLC
+      con.wallet_api_ptr->htlc_create("bob", "alice",
+            "3", "BOBCOIN", "HASH160", hash_str, preimage_string.size(), timelock, "Bob to Alice", true);
```

Comment:
Bob using the same `timelock` doesn't make sense since Alice would be able to redeem it after her own HTLC is expired.